src/actions/service.ts

```
@@ -43,26 +88,48 @@ export async function update(id: string, _: unknown, formData: FormData): Promis
+
+  const getServiceImage = async () => {
+    switch (service.kind) {
+      case "git": return await getLatestBuild();
+      case "database": return service.repo;
+    }
+  };
+
+  const image = await getServiceImage();
+  const providerVariables = service.providers
+    .map(provider => getProviderVariables(service.name, provider.name, provider.variables))
+    .reduce((bundle, current) => ({ ...bundle, ...current }), {});
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)